src/subgraphs/getData.ts

```diff
@@ -117,68 +139,85 @@ export class DataRecoveryGraph extends AbstractGraph<DataRecoveryState> {
       return {
         ...state,
         queryAttempts, 
-        finalResult: "Unable to generate a suitable query after 3 attempts. Here is the feedback message: " + state.feedbackMessage
+        finalResult: "Unable to generate a suitable query after 3 attempts. Here is the feedback message: " + state.feedbackMessage,
+        status: {type: 'error', error: { message: "Unable to generate a suitable query after 3 attempts" }}
```

Comment:
This is not using the type
I'd create a helper function which creates node states, we pass it the type, the message and returns the object that we know we can pass to status

If we want to change the output format from the AI to the frontend we just touch that function and it updates everywhere